src/NuGet.Core/NuGet.Packaging/PackageCreation/Authoring/IconValidation.cs

```diff
@@ -0,0 +1,50 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.IO;
+using NuGet.Packaging.Core;
+using NuGet.Packaging.PackageCreation.Resources;
+
+namespace NuGet.Packaging
+{
+    /// <summary>
+    /// Utility methods for package icon validation
+    /// </summary>
+    public class IconValidation
+    {
+        /// <summary>
+        /// Maximun Icon file size: 1 megabyte
+        /// </summary>
+        public const int MaxIconFileSize = 1024 * 1024;
+
+        /// <summary>
+        /// Validate the icon filesize
+        ///
+        /// Launches a PackagingException in case of an error
+        ///
+        /// This validation is intended to run at pre-packing time
+        /// </summary>
+        /// <remarks>This consumes the stream</remarks>
+        /// <param name="stream">Stream that points to the icon file</param>
+        /// <exception cref="PackagingException">When the icon file size is:
+        /// 1) Greater than the maximum icon file size <see cref="MaxIconFileSize"/>
+        /// 2) Less than zero, indicating an error reading the icon file
+        /// 3) Zero, indicating an empty file
+        /// </exception>
+        public static void ValidateIconFileSize(Stream stream)
```

Comment:
This doesn't have to be in an icon specific class. 

You are validating the length of a stream, you only need that utility method. 

Then the rest of the logic can just be done in the package builder.

Reply:
I see. 

I'd still consider extracting the stream specific details to separate classes. 
I get wanting to the validation utilities to be self ctaonined but right now it's not doing much :)

Reply:
Ok, I will move all the validation code to `PackageBuilder` class